src/docfx/config/PreloadConfig.cs

```diff
@@ -57,27 +57,16 @@ internal class PreloadConfig
         /// </summary>
         public string? DocsRepositoryOwnerName { get; init; }
 
-        public Action<HttpRequestMessage> GetCredentialProvider()
+        public CredentialHandler GetCredentialHandler()
         {
-            var rules = Http.OrderByDescending(pair => pair.Key, StringComparer.Ordinal).ToArray();
-
-            return message =>
+            return new Lazy<CredentialHandler>(() =>
```

Comment:
>new Lazy<CredentialHandler> [](start = 19, length = 27)

what is the meaning for this `Lazy` as we immediately invoke `.Value`?

Reply:
To get the same instance when it gets called multiple times